src/praisonai-bot/praisonai_bot/bots/_redis_turn_lock.py

```
@@ -0,0 +1,294 @@
+        # waiters cheaply and is the fail-open fallback when Redis is down.
+        self._local_lock = self._lock._local.get(self._local_key)
+        await self._local_lock.acquire()
+        ttl_ms = max(1, int(self._lock._ttl * 1000))
+        poll = self._lock._poll_interval
+        deadline = time.monotonic() + self._lock._ttl
+        while True:
+            try:
+                ok = await self._lock._client.set(
+                    self._redis_key, self._token, nx=True, px=ttl_ms
+                )
+            except Exception as e:  # Redis outage -> fail open on local lock.
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)